realm/realm-library/src/main/java/io/realm/BaseRealm.java

```diff
@@ -281,6 +283,40 @@ public void refresh() {
     }
 
     /**
+     * Blocks the current thread until transactions are committed by other threads to this Realm or
```

Comment:
Reading this. It still isn't clear to me the exact behaviour of this. Would the following be more accurate:

```
Blocks the current thread until new changes to the Realm are available or {@link #stopWaitForChange} is called.
```

From my understanding the method would return immediately if changes are available, which makes it a bit strange to to talk about committing from other thread?